rust/parquet/Cargo.toml

```diff
@@ -29,20 +29,29 @@ build = "build.rs"
 edition = "2018"
 
 [dependencies]
-parquet-format = "~2.6"
+parquet-format = "2.6.1"
```

Comment:
parquet-format 2.6.0 depends on thrift 0.12, which declares all its dependencies with ~ which is overly restrictive. Specifically, recent versions of the bincode crate declares a dependency on byteorder ^1.3 but thrift 0.12 declares a dependency on byteorder ~1.2 which is a requirement that cannot be resolved.

parquet-format 2.6.1 bumps to thrift 0.13 which no longer uses ~ for its dependencies.

Reply:
Understood about the need to update to thrift 0.13. However I thought the semver `~2.6` is equivalent to`>=2.6.0 < 2.7.0` so it will automatically pick up 2.6.1. Is that not the case?

Reply:
I don't know for certain. Things _might_ work with `~2.6`, but that does not forbid Cargo from selecting `2.6.0` and I've recently been fighting a lot with a few problems that arise because `cargo publish` runs `cargo update` and crate maintainers in general tend to update their lockfiles often.

Here's an example of a similar (but not exactly the same) problem, where the dependency specification includes things that do and don't compile: https://github.com/achanda/ipnetwork/issues/132

My perspective is that I'd rather be safe than sorry, but if you'd rather be conservative in what gets changed in the project I understand and I have no problem with opening a separate PR when this causes problems.

Reply:
Cool. Didn't know that. I think it should be fine to change to 2.6.1.